darts/ad/detectors/iqr_detector.py

```diff
@@ -0,0 +1,102 @@
+"""
+Interquartile Range (IQR) Detector
+-----------------
+
+Flags anomalies that are beyond the IQR (between the third and the first quartile)
+of historical data by some factor of it's difference (typically 1.5).
+This is similar to a threshold-based detector, but the thresholds are
+computed as distances from the IQR of historical data when the detector is fitted.
+"""
+
+from typing import Sequence, Union
+
+import numpy as np
+
+from darts.ad.detectors.quantile_detector import QuantileDetector
+from darts.ad.detectors.threshold_detector import ThresholdDetector
+from darts.logging import get_logger, raise_log
+from darts.timeseries import TimeSeries
+
+logger = get_logger(__name__)
+
+
+class IQRDetector(QuantileDetector):
+    def __init__(self, scale: Union[Sequence[Union[float, None]], float] = 1.5) -> None:
+        """IQR Detector
+
+        Flags values that lie outside of the interquartile range (IQR)
+        by more than a certain factor of IQR's value as anomalies.
+        The factor is passed in the `scale` parameter.
+
+        If a single value is provided for `scale`,
+        this same value will be used across all components of the series.
+
+        If a sequences of values is given for the `scale` parameter,
+        it's length must match the dimensionality of the series passed.
+        Unless it's length is 1, then it is treated as a single value.
+        Infinity may be used to disable flagging on that dimension.
+
+        Parameters
+        ----------
+        scale
+            (Sequence of) scale(s) used to indicate what distance from the IQR constitutes an anomaly.
+            Defaults to `1.5`. Must be non-negative. If a sequence, must match the dimensionality of the series
+            this detector is applied to.
+        """
+
+        # Parent QuantileDetector will compute Q1 and Q3 thresholds
+        super().__init__(low_quantile=0.25, high_quantile=0.75)
+
+        self.scale = np.array(scale)
```

Comment:
let's make sure we convert all `scale` values into a 1d array 
```suggestion
        scale = [scale] if isinstance(scale, float) else scale
        self.scale = np.array(scale)
```

then you can raise an error if `self.scale.ndim != 1`. That simplifies things for the downstream operations as we can assume a fixed dimensionality.